batched/dense/impl/KokkosBatched_HostLevel_Gemm_Impl.hpp

```diff
@@ -0,0 +1,689 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#ifndef __KOKKOSBATCHED_HOSTLEVEL_GEMM_IMPL_HPP__
+#define __KOKKOSBATCHED_HOSTLEVEL_GEMM_IMPL_HPP__
+#include "Kokkos_Core.hpp"         // KOKKOS_INLINE_FUNCTION
+#include "KokkosBatched_Util.hpp"  // Trans, BatchLayout
+#include <KokkosBatched_Gemm_Handle.hpp>
+#include <KokkosKernels_ExecSpaceUtils.hpp>
+#include <KokkosKernels_Error.hpp>
+
+namespace KokkosBatched {
+namespace Impl {
+/********************* BEGIN forward declarations *********************/
```

Comment:
Feel free to remove all the forward declarations and to include the impl headers up here.